### DISCUSSION TITLE
VS Code debugger not working for me

### DISCUSSION THREAD

**Discussion by james-cnz** [picoe/Eto#2518] [General]
Hello,

I recently installed Eto.Forms (on Debian 11 with .NET 6), and the debugger in VS Code was working for me at first, but now it's giving an error:

Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'dotnet_eto_cs_test, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'dotnet_eto_cs_test, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'

I don't know what's changed, perhaps a software update?  Does anyone know what the problem might be?

Thanks,
James

**Comment by cwensley** [comment]
Latest VS Code (1.80.0) is still working for me on ubuntu.  I'd check your configurations and the build outputs.

**Comment by james-cnz** [comment]
I've tried playing with the configurations, but I don't really know what I'm doing, and there doesn't seem to be anything in the Eto.Forms docs about it.  There's a warning relating to the WPF project, but I don't suppose it's relevant.  I get the problem with a new app created from the template.  I'm just doing this:

Create a new directory
Run dotnet new etoapp
Open with VS Code
Accept the notifications
- There are unresolved dependencies. Please execute the restore command to continue.
- Required assets to build and debug are missing from 'dotnet_eto_test'. Add them? (Choosing the GTK project)
And click on start debugging

I've managed to work around it, though, by combining the stuff from the GTK directory with the stuff in the platform-independent directory, and getting rid of the other directories.

**Comment by cwensley** [comment]
@james-cnz what does your `launch.json` file look like?  It really depends on how you have that set up for debugging in VS Code.

**Comment by james-cnz** [comment]
Thanks.  This is my launch.json file:

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": ".NET Core Launch (console)",
            "type": "coreclr",
            "request": "launch",
            "preLaunchTask": "build",
            "program": "${workspaceFolder}/MyApp.Gtk/bin/Debug/net6.0/MyApp.Gtk.dll",
            "args": [],
            "cwd": "${workspaceFolder}/MyApp.Gtk",
            "console": "internalConsole",
            "stopAtEntry": false
        },
        {
            "name": ".NET Core Attach",
            "type": "coreclr",
            "request": "attach"
        }
    ]
}
```

**Comment by cwensley** [comment]
Looks fine to me.. though I'm not sure where `dotnet_eto_cs_test` is coming in the picture..   Perhaps zipping it up would help me understand what the problem is here.. or recording a video..   Are you sure you're loading the correct directory in VS Code?

Otherwise there's not much information for me to go on here.

**Comment by james-cnz** [comment]
Oh, sorry, I've deleted the project and created a new one from the template several times.  "dotnet_eto_cs_test" is the project name I used most often.  I switched to "MyApp" when I sent the file, so it matched the example on the website.  I've zipped and attached a new project generated from the template.  The comments in the launch.json are slightly different.  I think it might depend on whether I generate the launch.json file through the notifications or the run & debug menu.  The content's the same though, and I get the same error message.
[MyApp.zip](https://github.com/picoe/Eto/files/12206473/MyApp.zip)
